test/net.js

```diff
@@ -432,16 +430,28 @@ tape('timeout', function (t) {
       socket.destroy()
     })
     socket.on('close', function () {
-      clientClosed = true
-      done()
+      close.pass('client closed')
     })
   })
 
-  function done () {
-    if (--missing) return
+  await close
+  server.close()
+})
+
+test('exception in connection listener', async function (t) {
+  t.plan(1)
+
+  const server = utp.createServer(function (socket) {
+    socket.destroy()
+    throw new Error('disconnect')
+  })
+
+  process.once('uncaughtException', () => {
```

Comment:
@davidmarkclements Brittle sets up an uncaught exception capture callback so this event isn't emitted. Would it be possible to have Brittle emit it and bail out if it was handled by a listener?

Reply:
yes absolutely, would you be okay to make that PR to brittle?